google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```diff
@@ -1577,4 +1583,140 @@ public URL generateV4GPutbjectSignedUrl(String bucketName, String objectName)
+    // String serviceAccountEmail = "service-account@iam.gserviceaccount.com";
+    ServiceAccount account = ServiceAccount.of(serviceAccountEmail);
+    HmacKey hmacKey = storage.createHmacKey(account);
+
+    String secret = hmacKey.getSecretKey();
+    HmacKeyMetadata metadata = hmacKey.getMetadata();
+
+    System.out.println("The Base64 encoded secret is: " + secret);
+    System.out.println("Do not miss that secret, there is no API to recover it.");
+    System.out.println("The HMAC key metadata is:");
+    System.out.println("ID: " + metadata.getId());
+    System.out.println("Access ID: " + metadata.getAccessId());
+    System.out.println("Project ID: " + metadata.getProjectId());
+    System.out.println("Service Account Email: " + metadata.getServiceAccount().getEmail());
+    System.out.println("State: " + metadata.getState().toString());
+    System.out.println("Time Created: " + new Date(metadata.getCreateTime()).toString());
+    System.out.println("Time Updated: " + new Date(metadata.getUpdateTime()).toString());
+    System.out.println("ETag: " + metadata.getEtag());
+    // [END storage_create_hmac_key]
+    return hmacKey;
+  }
+
+  /** Example of retrieving the metadata of an existing HMAC key. * */
+  public HmacKeyMetadata getHmacKey(String accessId) throws StorageException {
+    // [START storage_get_hmac_key]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The access ID of the HMAC key, e.g. "GOOG0234230X00"
+    // String accessId = "GOOG0234230X00";
+    HmacKeyMetadata metadata = storage.getHmacKey(accessId);
+
+    System.out.println("The HMAC key metadata is:");
+    System.out.println("ID: " + metadata.getId());
+    System.out.println("Access ID: " + metadata.getAccessId());
+    System.out.println("Project ID: " + metadata.getProjectId());
+    System.out.println("Service Account Email: " + metadata.getServiceAccount().getEmail());
+    System.out.println("State: " + metadata.getState().toString());
+    System.out.println("Time Created: " + new Date(metadata.getCreateTime()).toString());
+    System.out.println("Time Updated: " + new Date(metadata.getUpdateTime()).toString());
+    System.out.println("ETag: " + metadata.getEtag());
+    // [END storage_get_hmac_key]
+    return metadata;
+  }
+
+  /** Example of deleting an existing inactive HMAC key. * */
+  public void deleteHmacKey(String accessId) throws StorageException {
+    // [START storage_delete_hmac_key]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The access ID of the HMAC key, e.g. "GOOG0234230X00"
+    // String accessId = "GOOG0234230X00";
+    HmacKeyMetadata metadata = storage.getHmacKey(accessId);
+    storage.deleteHmacKey(metadata);
+
+    System.out.println("The key is deleted, though it may still appear in getHmacKeys() results.");
```

Comment:
That only happens if showDeletedKeys is set to true right? Maybe specify that

Reply:
Added. Nice catch!